app/queries/update_post_mutator.rb

```diff
@@ -3,11 +3,11 @@ class UpdatePostMutator < ApplicationQuery
 
   property :id
   property :body, validates: { length: { minimum: 1, maximum: 10_000 }, presence: true }
-
+  property :edit_reason
   def update_post
     Post.transaction do
-      post.text_versions.create!(value: post.body, user: post.creator, edited_at: post.updated_at)
-      post.update!(body: body, editor: current_user)
+      post.text_versions.create!(value: post.body, user: post.creator, edited_at: post.updated_at, reason: post.edit_reason)
+      post.update!(body: body, editor: current_user, edit_reason: edit_reason)
```

Comment:
Do not store text value received from _user-land_ directly into the DB. Let's clean that up a bit:

```ruby
post.update!(body: body, editor: current_user, edit_reason: sanitized_edit_reason)

# ...

# Returns stripped edit reason, or nil.
def sanitized_edit_reason
  stripped_reason = edit_reason.strip
  stripped_reason.presence
end
```